### PR TITLE
Finalize Subscription Payment 

### PR DESCRIPTION
**Pull request by Ashlynn223**
created basic flow that allows a user to subscribe to a yearly or monthly payment

### CODE REVIEW

**Comment by Jake-Mitchell** [comment]
This is it! from the ashes of our spike and learning about Stripe API, the growth of a subscription has been born 🔥🌱

File: cha-ching-app/src/SubscriptionPage.js [hunk 1]
```diff
@@ -25,9 +21,12 @@ const SubscriptionPage = () => {
       }),
     }; 
 
-    fetch("http://localhost:3001/create-product", requestOptions)
+    fetch("http://localhost:3001/create-incomplete-subscription", requestOptions)
       .then((res) => res.json())
-      .then(data => console.log("create product data", data))
+      .then(data => {
+        console.log("client secret", data.latest_invoice.payment_intent.client_secret)
```
Comment (angepol) [inline cha-ching-app/src/SubscriptionPage.js:27]: You could remove this console.log

File: cha-ching-app/src/index.js [hunk 2]
```diff
@@ -5,10 +5,11 @@ import App from "./App";
 import reportWebVitals from "./reportWebVitals";
 
 const root = ReactDOM.createRoot(document.getElementById("root"));
+//TODO: look into if there are any potential cons in removing strict mode
```
Comment (Jake-Mitchell) [inline cha-ching-app/src/index.js:8]: yes! We can even make a ticket for this one and put it in the icebox

File: cha-ching-app/src/CheckoutForm.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ function CheckoutForm() {
         },
       })
       .then((data) => console.log("What am I (data):", data))
-      .catch((data) => console.log("I am an error"));
+      .catch((err) => console.log("I am an error", err));
```
Comment (angepol) [inline cha-ching-app/src/CheckoutForm.js:20]: Nice!

**Review by angepol — APPROVED** [review]

**Pull request merged**